services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -115,17 +115,18 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
       Future.failed(VersioningException.RepositoryException("Could not create a repository for deleted project."))
     else {
       val createRepoUrl: String = s"${config.url}projects"
+      val postProject = PostProject(name = project.name)
       httpClient
-        .post[Project, Project](url = createRepoUrl, headers = config.token, payload = project)
-        .map(
-          resp =>
-            if (resp.status != HttpStatusCodes.Created)
-              Left(
-                VersioningException
-                  .RepositoryException(s"The repository was not created. Response status: ${resp.status}")
-              )
-            else Right(project.withRepository(Some(s"${config.idPath}${project.projectId.value}")))
-        )
+        .post[RepositoryId, PostProject](url = createRepoUrl, headers = config.token, payload = postProject)
```

Comment:
Interesting change. Does it work? :) Did it work before your changes?
Before your changes, this code send to GitLab an object with the following fields 
`projectId, ownerId, name, active, repository, visibility`
And expected the object with the same fields in return.
Now, we are sending an object with only one field `name`. This could be okay if GitLab never wanted a full `Project` object and wanted only it's name . But we also expect that GitLab will return an object with one field `id`. We never wanted this field before. Does GitLab API return it?
We can't just change the signature of the request and expect that it will work automatically :)
It is possible that this request never actually worked and now it is fixed. But I have to explicitly ask about it.

Reply:
@myazinn https://jirapct.epam.com/jira/browse/EPMLSTRCMW-176 :)

Reply:
@GrigoriiBerezin I guess I should have read the task earlier :)
@manvicha thank you for such a detailed answer. In this case, I have no more questions.